creme/creme_core/views/entity_filter.py

```diff
@@ -270,48 +274,68 @@ def get(self, request, *args, **kwargs):
-            default=EF_USER,  # TODO: make mandatory ?
-        )
+class EntityFilterUserEnumerator(UserEnumerator):
+    def __init__(
+        self,
+        field: Field,
+        filter_type=EF_USER,
+        search_fields=None,
+        limit_choices_to=None
+    ):
+        self.efilter_registry = self.get_efilter_registry(filter_type)
+        super().__init__(field, search_fields, limit_choices_to)
 
+    def get_efilter_registry(self, filter_type):
         try:
-            return entity_filter_registries[ftype]
-        except KeyError as e:
-            raise Http404('Invalid filter type') from e
-
-    def get_operands(self):
-        user = self.request.user
+            return entity_filter_registries[filter_type]
+        except KeyError:
+            raise BadRequestError(f'Unknown entity filter type {filter_type}')
 
-        for operand in self.get_registry().operands(user):
+    def get_user_operands(self, user):
+        for operand in self.efilter_registry.operands(user):
             if issubclass(operand.model, User):
                 yield operand
 
-    def get(self, request, *args, **kwargs):
-        sort_key = collator.sort_key
+    def to_python(self, user, values):
+        operands = [o.type_id for o in self.get_user_operands(user)]
 
-        def choice_key(c):
-            return sort_key(c[1])
+        op_ids, pks = partition(lambda v: v in operands, values)
+        return list(op_ids) + list(self._queryset(user).filter(pk__in=pks))
 
-        # TODO: return group for teams & inactive users (see UserEnumerator)
-        #       => fix the JavaScript side (it concatenates the group label at the end)
-        return self.response_class(
-            [
-                *sorted(
-                    ((op.type_id, op.verbose_name) for op in self.get_operands()),
-                    key=choice_key,
-                ),
-                *sorted(
-                    ((e.id, str(e)) for e in User.objects.all()),
-                    key=choice_key,
-                ),
-            ],
-            safe=False,  # Result is not a dictionary
+    def choices(self, user, *, term=None, only=None, limit=None):
+        sort_key = collator.sort_key
+        choices = [
+            {
+                'value': op.type_id,
+                'label': op.verbose_name,
+            } for op in self.get_user_operands(user)
+        ]
+        choices.sort(key=lambda d: sort_key(d['label']))
+
+        choices.extend(
+            super().choices(user, term=term, only=only)
         )
+
+        return choices[:limit] if limit else choices
+
+
+class UserChoicesView(ChoicesView):
+    filter_type_arg = 'filter_type'
+
+    def get_enumerator(self):
+        model = self.get_ctype().model_class()
+        field_name = self.get_field_name()
+
+        try:
+            field = model._meta.get_field(field_name)
+            return EntityFilterUserEnumerator(field, filter_type=self.filter_type)
```

Comment:
Can it raise a `FieldDoesNotExist`? If not, get this line out of the `try..except...` block.